components/ciphertool/src/main/java/org/wso2/ciphertool/utils/Utils.java

```diff
@@ -206,8 +206,11 @@ public static void writeToSecureConfPropertyFile() {
         properties.setProperty(Constants.SecureVault.KEYSTORE_KEY_SECRET_PROVIDER,
                                Constants.SecureVault.CARBON_DEFAULT_SECRET_PROVIDER);
         // This property is referred by the FileBaseSecretRepository for decryption.
-        properties.setProperty(Constants.SecureVault.SECRET_FILE_ALGORITHM,
-                System.getProperty(Constants.CIPHER_TRANSFORMATION_SYSTEM_PROPERTY));
+        String algorithm = System.getProperty(Constants.CIPHER_TRANSFORMATION_SYSTEM_PROPERTY);
+        if (algorithm != null) {
+            properties.setProperty(Constants.SecureVault.SECRET_FILE_ALGORITHM,
+                    System.getProperty(Constants.CIPHER_TRANSFORMATION_SYSTEM_PROPERTY));

```

Comment:
we can use "algorithm" here without evaluating again

Reply:
fixed.